internal/core/api/workspace.josh.hcl

```diff
@@ -185,6 +204,7 @@ struct "process-description" {
   field "id" "string" {}
   field "provider" "string" {}
   field "name" "string" {}
+  field "spec" "interface{}" {}
```

Comment:
Is the intention here that this should be of type `process.Spec` from the `providers/unix/components/process` package? Maybe needs a comment, as this is a shortcoming of our code generator now.

Reply:
The concrete type of this is going to depend on the component type. For a unix process, it is a `process.Spec`, but for a docker container, it would be a different type. Deserialization can happen in each branch of the switch where we build components from the serialized data.

Reply:
In that case, I think this is buggy for Docker use cases, as we're going to serialize the JOSH, which will call MarshalJSON but really we want MarshalYAML. The type of the spec should just be `string`.

Reply:
Okay. That kind of sucks that we have to unmarshal on use rather than on deserialization, but that makes sense. I'll change it and unmarshal it where I need it.

Reply:
Yeah, I've had some internal debate with myself on this topic. Ultimately, I decided that using a text format for the spec was the right thing to do because it lets us use shell scripts, docker compose yaml, terraform or hcl, etc directly without having to invent our own serialization formats. This of course does mean that any editing code paths or whatever will need to serialize/deserialize and lack generic editing of properties etc, but I don't think it is realistic for us to do generic editing of things like docker yaml, k8s yaml, etc which we don't control.

In general, treating it as an opaque string and marshalling/unmarshalling is maximally general, with all the pros and cons that usually brings along. For now, let's stick with that, and once we have more experience with an increasing number of components, we can decide if we want to invent a spec format that gives us the flexibility we need. For instance, we may want to use HCL so that we can have expressions or similar in there when we get to dynamically reconfigurable service constellations and need to include/exclude/change environment variables depending on what's running on host, docker, or cloud.